projects/com.oracle.truffle.llvm.parser.base/src/com/oracle/truffle/llvm/parser/base/facade/NodeFactoryFacade.java

```diff
@@ -108,12 +107,13 @@
     /**
      * Creates an intrinsic for a <code>@llvm.*</code> function.
      *
-     * @param functionName the name of the intrinsic function starting with <code>@llvm.</code>
+     * @param declaration the function declaration of the function from which the intrinsic is
+     *            called
      * @param argNodes the arguments to the intrinsic function
-     * @param functionDef the function definition of the function from which the intrinsic is called
+     * @param argCount number of arguments (could differ from declaration arg count)
```

Comment:
In which cases can `argCount` differ from the declaration arg count? Is `argCount == argNodes.length`? Then we could remove the `argCount` again.